tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Whatis.java

```diff
@@ -0,0 +1,172 @@
+package org.unicode.cldr.web.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.unicode.cldr.util.CLDRFile;
+import org.unicode.cldr.util.CLDRLocale;
+import org.unicode.cldr.util.StandardCodes;
+import org.unicode.cldr.web.CookieSession;
+import org.unicode.cldr.web.SurveyMain;
+import org.unicode.cldr.web.api.WhatisResponse.StandardCodeResult;
+import org.unicode.cldr.web.api.WhatisResponse.XpathResult;
+
+@Path("/whatis")
+public class Whatis {
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "Look up a code",
+        description = "Searches for codes containing the given string, with the given base locale")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "200",
+                description = "Look up a code",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = WhatisResponse.class))),
+        })
+    public Response getWhatis(
+        @Parameter(required = true, example = "jgo", schema = @Schema(type = SchemaType.STRING)) @QueryParam("q") String q,
+        @Parameter(required = true, example = "en_US", schema = @Schema(type = SchemaType.STRING)) @QueryParam("loc") String loc) {
+
+        q = q.trim();
+        if (q.isEmpty()) {
+            Map<String, String> r = new TreeMap<>();
+            r.put("err", "empty query");
+            return Response.ok(r).build();
+        }
+        WhatisResponse what = new WhatisResponse(q);
+        if (q.startsWith("\\u") || q.startsWith("=")) {
+            searchPathsAndValues(what); // doesn't use loc
+        } else {
+            searchStandardCodesExact(what, loc);
+            searchStandardCodesFull(what, loc);
+        }
+        return Response.ok().entity(what).build();
+    }
```

Comment:
@srl295 I've imitated LoginResponse.java for `entity(what)` here and `@Schema(implementation = WhatisResponse.class)` above